.github/workflows/main.yml

```diff
@@ -0,0 +1,22 @@
+name: CI
+
+on:
+  push:
+    branches: [main]
```

Comment:
I'm going to skip this for now. It should pass in the PR but if we just implement that this way, it will run at the same time as deployment and if it fails it wouldn't block it. Need to rework the deployment workflow for this to make sense.